pingcap/tidb
# server: log last executed prepared stmt when panic

### What problem does this PR solve? 

current prepare execute meet panic will output log like this:

```
[2019/10/17 00:21:55.082 +08:00] [ERROR] [conn.go:600] ["connection running loop panic"] [conn=1934] [lastCmd="\u0001\u0000\u0000\u0000\u0000\u0001\u0000\u0000\u0000\u0000"] [err="\"invalid memory address or nil pointer dereference\""] [stack="goroutine 53291527 [running]:\ngithub.com/pingcap/tidb/server.(*clientConn).Run.func1(0x23006c0, 0xc02f5ecdb0, 0xc00c46f860)\n\t/root/go/src/github.com/pingcap/tidb/server/conn.go:598 +0xee\npanic(0x1e1f600, 0x30b5970)\n\t/usr/local/go/src/runtime/panic.go:679 +0x1b2\ngithub.com/pingcap/tidb/util/chunk.Row.IsNull(...)
```
it's hard to identify which sql cause problem.

### What is changed and how it works?

make com_execute / com_execute output right sql.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
```
modify code to panic and see log
```

Code changes

 - impl


Side effects

 - n/a

Related changes

 - Need to cherry-pick to the release branch

Release note

 -  log last executed prepared stmt when panic


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/12905)
